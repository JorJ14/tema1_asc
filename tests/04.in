{
    "products": {
        "id1": {
            "product_type": "Coffee",
            "name": "Indonezia",
            "acidity": 5.05,
            "roast_level": "MEDIUM",
            "price": 1
        },
        "id2": {
            "product_type": "Coffee",
            "name": "Brasil",
            "acidity": 5.09,
            "roast_level": "MEDIUM",
            "price": 7
        },
        "id3": {
            "product_type": "Tea",
            "name": "Wild Cherry",
            "type": "Black",
            "price": 5
        }
    },
    "producers": [
        {
            "name": "prod1",
            "products": [
                [
                    "id2",
                    3,
                    0.1
                ],
                [
                    "id1",
                    1,
                    0.31
                ]
            ],
            "republish_wait_time": 0.27
        },
        {
            "name": "prod2",
            "products": [
                [
                    "id3",
                    2,
                    0.08
                ],
                [
                    "id1",
                    1,
                    0.36
                ]
            ],
            "republish_wait_time": 0.29
        },
        {
            "name": "prod3",
            "products": [
                [
                    "id3",
                    2,
                    0.2
                ],
                [
                    "id1",
                    3,
                    0.27
                ]
            ],
            "republish_wait_time": 0.12
        },
        {
            "name": "prod4",
            "products": [
                [
                    "id2",
                    1,
                    0.33
                ],
                [
                    "id3",
                    3,
                    0.37
                ],
                [
                    "id1",
                    1,
                    0.3
                ]
            ],
            "republish_wait_time": 0.19
        },
        {
            "name": "prod5",
            "products": [
                [
                    "id3",
                    2,
                    0.14
                ],
                [
                    "id1",
                    2,
                    0.3
                ],
                [
                    "id2",
                    1,
                    0.12
                ]
            ],
            "republish_wait_time": 0.25
        }
    ],
    "consumers": [
        {
            "name": "cons1",
            "retry_wait_time": 0.13,
            "carts": [
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 5
                    },
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 4
                    },
                    {
                        "type": "add",
                        "product": "id2",
                        "quantity": 1
                    },
                    {
                        "type": "remove",
                        "product": "id3",
                        "quantity": 1
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 4
                    },
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 1
                    },
                    {
                        "type": "add",
                        "product": "id2",
                        "quantity": 5
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 1
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 1
                    },
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 5
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 1
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 2
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 4
                    },
                    {
                        "type": "remove",
                        "product": "id3",
                        "quantity": 2
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 4
                    },
                    {
                        "type": "add",
                        "product": "id2",
                        "quantity": 3
                    },
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 1
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 1
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 3
                    },
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 1
                    },
                    {
                        "type": "add",
                        "product": "id2",
                        "quantity": 2
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 2
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 2
                    },
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 1
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id3",
                        "quantity": 2
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id2",
                        "quantity": 1
                    },
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 5
                    }
                ]
            ]
        }
    ],
    "marketplace": {
        "queue_size_per_producer": 13
    }
}
