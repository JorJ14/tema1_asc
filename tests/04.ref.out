cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Brasil', price=7, acidity=5.09, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
cons1 bought Tea(name='Wild Cherry', price=5, type='Black')
